### PR TITLE
convert std::cout and std::cerr to std::println()

### PR DESCRIPTION
**Pull request by Toni500github**
### Describe your PR, what does it fix/add?
follow up from #8034 
`std::print()` is easier to use and read, it also has better performance and reduces boilerplate code https://vitaut.net/posts/2023/print-in-cpp23/

### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
So I have almost converted every instance of `std::cout`, I only left some for example where `hyprpm` prints the warning message when adding a plugin and there were too many Colors, so I left those for the sake of readability.

I have added `hyprpm/src/helpers/StringUtils.hpp` which is useful to replace the boilerplate such as
```c++
std::cout << Colors::BLUE << "[v] " << Colors::RESET << "version returned: " << HLVERCALL << "\n";
```
with
```c++
std::println("{}", verboseString("version returned: {}", HLVERCALL));
```

### Is it ready for merging, or does it need work?
Yeah, hope I didn't do too much or some things must be in a separeted MR. This took me the whole day. Did some tests and seems good so far

### CODE REVIEW

**Comment by Toni500github** [comment]
is the indetion good? 1st: current, 2nd: my MR
image

File: hyprpm/src/core/PluginManager.cpp
Comment (vaxerski) [inline hyprpm/src/core/PluginManager.cpp:527]: considering you made those take a format string wouldn't it be nice to also get rid of those and just `{}` them? :)

**Review by vaxerski — COMMENTED** [review]
also if you could split those into 3 commits:
 - hyprland
 - hyprpm
 - hyprctl
 
then I could just rebase it on main.

**Comment by vaxerski** [comment]
> is the indetion good? 1st: current, 2nd: my MR

ye thats whatever both look alr

File: hyprpm/src/core/PluginManager.cpp
Comment (Toni500github) [reply hyprpm/src/core/PluginManager.cpp:527]: With pleasure

Comment (Toni500github) [reply hyprpm/src/core/PluginManager.cpp:527]: So make them 
```c++
progress.printMessageAbove(verboseString("installation will run: {}", cmd));
```
Instead right?

**Comment by Toni500github** [comment]
> also if you could split those into 3 commits:
>  - hyprland
>  - hyprpm
>  - hyprctl
>  
> then I could just rebase it on main.

Sorry wdym?
Like re-creating these 5 commits in 3 and separated by categories?

File: hyprpm/src/core/PluginManager.cpp
Comment (vaxerski) [reply hyprpm/src/core/PluginManager.cpp:527]: ya

**Comment by vaxerski** [comment]
> Sorry wdym?
Like re-creating these 5 commits in 3 and separated by categories?

Squash all commits into one, then split that one commit into three, each changing only one component (hyprland, hyprctl, hyprpm)

so it's
 - hyprland: convert std::cout and std::cerr to std::println
 - hyprpm: convert std::cout and std::cerr to std::println
 - hyprctl: convert std::cout and std::cerr to std::println

**Comment by Toni500github** [comment]
> > Sorry wdym?
> > Like re-creating these 5 commits in 3 and separated by categories?
> 
> Squash all commits into one, then split that one commit into three, each changing only one component (hyprland, hyprctl, hyprpm)
> 
> so it's
> 
> * hyprland: convert std::cout and std::cerr to std::println
> * hyprpm: convert std::cout and std::cerr to std::println
> * hyprctl: convert std::cout and std::cerr to std::println

done

File: hyprpm/src/core/PluginManager.cpp
Comment (Toni500github) [inline hyprpm/src/core/PluginManager.cpp:164]: ah dammmit, should I rebase everything again or you could just fix it in small commit?

Comment (Toni500github) [inline hyprpm/src/core/PluginManager.cpp:164]: ah dammmit, should I rebase everything again or you could just fix it in small commit?

**Comment by Toni500github** [comment]
btw bump `VERSION` to 0.44.1

**Comment by vaxerski** [comment]
> btw bump VERSION to 0.44.1

no

File: hyprpm/src/progress/CProgressBar.cpp
Comment (vaxerski) [inline hyprpm/src/progress/CProgressBar.cpp:28]: why did you make this a member? what if it changes?

**Comment by Toni500github** [comment]
let me know if there is anything else I missed or should changes, so I can rebase into 3 commits again

**Comment by Toni500github** [comment]
image

image
also fixed some status printing generally

File: hyprpm/src/progress/CProgressBar.cpp
Comment (Toni500github) [reply hyprpm/src/progress/CProgressBar.cpp]: You right, reverted

**Review by vaxerski — APPROVED** [review]
lgtm thanks!

**Comment by vaxerski** [comment]
image
can you squash those again so that there are only 3 commits :P

**Comment by Toni500github** [comment]
yeah ofc, was just waiting for approval

**Comment by Toni500github** [comment]
image

**Pull request merged**
